ddl/backfilling.go

```diff
@@ -708,20 +722,66 @@ func (b *backfillScheduler) adjustWorkerSize() error {
 		b.workers = b.workers[:workerCnt]
 		closeBackfillWorkers(workers)
 	}
+	if b.copReqSenderPool != nil {
+		b.copReqSenderPool.adjustSize(len(b.workers))
+	}
 	return injectCheckBackfillWorkerNum(len(b.workers))
 }
 
-func (b *backfillScheduler) handleCreateBackfillWorkerErr(err error) error {
-	if len(b.workers) == 0 {
-		return errors.Trace(err)
+func (b *backfillScheduler) initCopReqSenderPool() {
+	if b.tp != typeAddIndexWorker || b.reorgInfo.Job.ReorgMeta.ReorgTp != model.ReorgTypeLitMerge ||
+		b.copReqSenderPool != nil || len(b.workers) > 0 {
+		return
+	}
+	indexInfo := model.FindIndexInfoByID(b.tbl.Meta().Indices, b.reorgInfo.currElement.ID)
+	if indexInfo == nil {
+		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender",
+			zap.Int64("table ID", b.tbl.Meta().ID), zap.Int64("index ID", b.reorgInfo.currElement.ID))
+		return
+	}
+	sessCtx, err := b.newSessCtx()
+	if err != nil {
+		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender", zap.Error(err))
+		return
+	}
+	copCtx := newCopContext(b.tbl.Meta(), indexInfo, sessCtx)
+	if copCtx == nil {
+		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender")
+		return
+	}
+	startTS, err := latestStartTS(sessCtx.GetStore())
+	if err != nil {
+		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender", zap.Error(err))
+		return
+	}
+	b.copReqSenderPool = newCopReqSenderPool(b.ctx, copCtx, startTS)
+}
+
+func latestStartTS(storage kv.Storage) (startTS uint64, err error) {
+	txn, err := storage.Begin()
+	if err != nil {
+		_ = txn.Rollback()
+		return 0, err
+	}
+	_ = txn.Rollback()
+	return txn.StartTS(), nil
+}
+
+func (b *backfillScheduler) canSkipError(err error) bool {
+	if len(b.workers) > 0 {
+		// The error can be skipped because the rest workers can handle the tasks.
```

Comment:
Could we add log before return?

Reply:
The error has been logged before calling `canSkipError`.